lib/storage-delegations.js

```diff
@@ -0,0 +1,193 @@
+/*!
+ * Copyright (c) 2019-2020 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const assert = require('assert-plus');
+const bedrock = require('bedrock');
+const database = require('bedrock-mongodb');
+const {promisify} = require('util');
+const {util: {BedrockError}} = bedrock;
+
+bedrock.events.on('bedrock-mongodb.ready', async () => {
+  await promisify(database.openCollections)(['zcap-delegation']);
+
+  // FIXME: what are indexes for this collection?
+
+  await promisify(database.createIndexes)([{
+    collection: 'zcap-delegation',
+    fields: {invocationTarget: 1, id: 1},
+    options: {unique: true, background: false}
+  }, {
+    collection: 'zcap-delegation',
+    fields: {controller: 1, id: 1},
+    options: {unique: true, background: false}
+  }]);
+});
+
+/**
+ * Inserts an delegation into storage.
+ *
+ * @param {string} controller - The ID of the entity storing the delegation.
```

Comment:
I haven't looked at usage of this lib. Do we need `controller` if we have `delegator`? Maybe we do to handle the case where `delegator` is just a key. But we don't have that for revocation storage below... so I'm not sure if we need it.

Reply:
Had the same question myself.

Reply:
Delegations has been removed.